cmd/containerd/command/config.go

```diff
@@ -65,13 +72,18 @@ func outputConfig(ctx gocontext.Context, config *srvconfig.Config) error {
 		}
 	}
 
-	// for the time being, keep the defaultConfig's version set at 1 so that
+	// for the time being, keep the defaultConfig's version set at 3 so that
 	// when a config without a version is loaded from disk and has no version
-	// set, we assume it's a v1 config.  But when generating new configs via
+	// set, we assume it's a v3 config.  But when generating new configs via
 	// this command, generate the max configuration version
 	config.Version = srvconfig.CurrentConfigVersion
 
-	return toml.NewEncoder(os.Stdout).SetIndentTables(true).Encode(config)
+	// Now merge on top of this, the actual user config that is present on disk
+	if err = srvconfig.MergeConfig(config, userConfig); err != nil {
```

Comment:
I'm trying to under the role the Merge is taking here. the plugin `Decode` is important to run on the user/default configuration as that will replace the `interface{}` initially decoded as `map[string]interface{}` into the struct provided by the plugin. I suspect when this `Merge` is called we may have the `to` case as the default plugin config struct and the `from` as the `map[string]interface{}` from dump (or nil from default)